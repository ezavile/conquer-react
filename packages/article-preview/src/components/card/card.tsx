import { FC } from 'react';

import { Author, BackgroundImage, Share } from 'components';

export interface CardProps {
  image: string;
  title: string;
  description: string;
  author: {
    avatar: string;
    name: string;
    publishedAt: string;
  };
}

export const Card: FC<CardProps> = ({
  image,
  title,
  description,
  author: { avatar, name, publishedAt },
}) => {
  return (
    <article className="flex flex-col bg-white rounded-md m-auto w-10/12 max-w-xl md:flex-row md:max-w-4xl">
      <BackgroundImage src={image} />
      <div className="px-8 pt-8 pb-4 relative">
        <h2 className="text-dark-grayish-blue mb-4">{title}</h2>
        <p className="text-desaturated-dark-blue mb-8">{description}</p>
        <div className="flex justify-between items-center">
          <Author avatar={avatar} name={name} publishedAt={publishedAt} />
          <Share />
        </div>
      </div>
    </article>
  );
};
