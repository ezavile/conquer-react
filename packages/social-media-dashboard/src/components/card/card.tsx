import styled from 'styled-components';

import { Theme } from 'models';

export const Card = styled.article`
  background-color: ${(props: { theme: Theme }) => props.theme.cardBg};
  border-radius: 0.25rem;
  padding: 1.5rem;
  position: relative;

  & > * {
    z-index: 1;
    position: relative;
  }

  &::after {
    content: '';
    position: absolute;
    left: 0;
    bottom: 0;
    right: 0;
    top: 0;
    background-color: transparent;
    transition: background-color ease 0.5s;
  }

  &:hover {
    &::after {
      background-color: ${(props: { theme: Theme }) => props.theme.cardHoverBg};
    }
  }
`;
