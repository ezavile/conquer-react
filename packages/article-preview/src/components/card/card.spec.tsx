import { render, RenderResult } from '@testing-library/react';
import { Card, CardProps } from './card';

describe('<Card />', () => {
  let rendered: RenderResult;

  const props: CardProps = {
    image: 'image.png',
    title:
      'Shift the overall look and feel by adding these wonderful touches to furniture in your home',
    description:
      'Ever been in a room and felt like something was missing? Perhaps it felt slightly bare and uninviting. Iâ€™ve got some simple tips to help you make any room feel complete.',
    author: {
      name: 'Michelle Appleton',
      avatar: 'avatar.png',
      publishedAt: '28 Jun 2020',
    },
  };

  beforeEach(() => {
    rendered = render(<Card {...props} />);
  });

  it('takes a snapshot', () => {
    expect(rendered.container).toMatchSnapshot();
  });
});
