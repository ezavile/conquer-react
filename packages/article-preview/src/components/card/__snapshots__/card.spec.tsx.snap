// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> takes a snapshot 1`] = `
<div>
  <article
    class="flex flex-col bg-white rounded-md m-auto w-10/12 max-w-xl md:flex-row md:max-w-4xl"
  >
    <div
      class="bg-center bg-cover rounded-t-md h-56 md:h-auto md:w-80 md:bg-left md:rounded-t-none md:rounded-l-md md:flex-shrink-0"
      style="background-image: url(image.png);"
    />
    <div
      class="px-8 pt-8 pb-4 relative"
    >
      <h2
        class="text-dark-grayish-blue mb-4"
      >
        Shift the overall look and feel by adding these wonderful touches to furniture in your home
      </h2>
      <p
        class="text-desaturated-dark-blue mb-8"
      >
        Ever been in a room and felt like something was missing? Perhaps it felt slightly bare and uninviting. Iâ€™ve got some simple tips to help you make any room feel complete.
      </p>
      <div
        class="flex justify-between items-center"
      >
        <div
          class="flex items-center whitespace-no-wrap"
        >
          <img
            alt=""
            class="w-12 h-12 rounded-full mr-4"
            src="avatar.png"
          />
          <div
            class="flex flex-col text-sm"
          >
            <span>
              <span
                class="sr-only"
              >
                Author:
              </span>
              <strong
                class="text-dark-grayish-blue font-bold"
              >
                Michelle Appleton
              </strong>
            </span>
            <span>
              <span
                class="sr-only"
              >
                Published at
              </span>
              <time
                class="text-grayish-blue"
                datetime="2020-28-6"
              >
                28 Jun 2020
              </time>
            </span>
          </div>
        </div>
        <div
          class="md:relative md:flex md:flex-col md:items-center"
        >
          <button
            aria-label="share"
            class="relative z-10 flex justify-center items-center rounded-full w-8 h-8 border-2 focus:outline-none focus:border-grayish-blue transition duration-100 border-light-grayish-blue bg-light-grayish-blue"
            type="button"
          >
            <svg
              class="transition duration-100"
              color="#6E8098"
              fill="currentColor"
              height="1.3rem"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(110, 128, 152);"
              viewBox="0 0 512 512"
              width="1.3rem"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M444.7 230.4l-141.1-132c-1.7-1.6-3.3-2.5-5.6-2.4-4.4.2-10 3.3-10 8v66.2c0 2-1.6 3.8-3.6 4.1C144.1 195.8 85 300.8 64.1 409.8c-.8 4.3 5 8.3 7.7 4.9 51.2-64.5 113.5-106.6 212-107.4 2.2 0 4.2 2.6 4.2 4.8v65c0 7 9.3 10.1 14.5 5.3l142.1-134.3c2.6-2.4 3.4-5.2 3.5-8.4-.1-3.2-.9-6.9-3.4-9.3z"
              />
            </svg>
          </button>
          <div
            class="absolute flex items-center rounded-b-md px-8 h-20 bottom-0 inset-x-0 bg-dark-grayish-blue transition duration-100 text-sm opacity-0 md:rounded-md md:left-auto md:h-auto md:justify-center md:py-3 md:mb-12 md:shadow-lg md:w-56 md:text-xs md:right-auto"
          >
            <span
              class="hidden md:block"
              style="width: 0px; height: 0px; bottom: -8px; position: absolute; border-left: 8px solid transparent; border-right: 8px solid transparent; border-top: 8px solid hsl(217, 19%, 35%);"
            />
            <strong
              class="text-grayish-blue uppercase mr-3 tracking-widest font-hairline"
            >
              share
            </strong>
            <ul
              class="flex"
            >
              <li
                class="flex mr-3"
              >
                <button
                  aria-label="share"
                  class="focus:outline-none focus:shadow-outline"
                  type="button"
                >
                  <svg
                    color="white"
                    fill="currentColor"
                    height="1.7em"
                    stroke="currentColor"
                    stroke-width="0"
                    style="color: white;"
                    viewBox="0 0 448 512"
                    width="1.7em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M400 32H48A48 48 0 0 0 0 80v352a48 48 0 0 0 48 48h137.25V327.69h-63V256h63v-54.64c0-62.15 37-96.48 93.67-96.48 27.14 0 55.52 4.84 55.52 4.84v61h-31.27c-30.81 0-40.42 19.12-40.42 38.73V256h68.78l-11 71.69h-57.78V480H400a48 48 0 0 0 48-48V80a48 48 0 0 0-48-48z"
                    />
                  </svg>
                </button>
              </li>
              <li
                class="flex mr-3"
              >
                <button
                  aria-label="share"
                  class="focus:outline-none focus:shadow-outline"
                  type="button"
                >
                  <svg
                    color="white"
                    fill="currentColor"
                    height="1.7em"
                    stroke="currentColor"
                    stroke-width="0"
                    style="color: white;"
                    viewBox="0 0 512 512"
                    width="1.7em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
                    />
                  </svg>
                </button>
              </li>
              <li
                class="flex "
              >
                <button
                  aria-label="share"
                  class="focus:outline-none focus:shadow-outline"
                  type="button"
                >
                  <svg
                    color="white"
                    fill="currentColor"
                    height="1.7em"
                    stroke="currentColor"
                    stroke-width="0"
                    style="color: white;"
                    viewBox="0 0 496 512"
                    width="1.7em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M496 256c0 137-111 248-248 248-25.6 0-50.2-3.9-73.4-11.1 10.1-16.5 25.2-43.5 30.8-65 3-11.6 15.4-59 15.4-59 8.1 15.4 31.7 28.5 56.8 28.5 74.8 0 128.7-68.8 128.7-154.3 0-81.9-66.9-143.2-152.9-143.2-107 0-163.9 71.8-163.9 150.1 0 36.4 19.4 81.7 50.3 96.1 4.7 2.2 7.2 1.2 8.3-3.3.8-3.4 5-20.3 6.9-28.1.6-2.5.3-4.7-1.7-7.1-10.1-12.5-18.3-35.3-18.3-56.6 0-54.7 41.4-107.6 112-107.6 60.9 0 103.6 41.5 103.6 100.9 0 67.1-33.9 113.6-78 113.6-24.3 0-42.6-20.1-36.7-44.8 7-29.5 20.5-61.3 20.5-82.6 0-19-10.2-34.9-31.4-34.9-24.9 0-44.9 25.7-44.9 60.2 0 22 7.4 36.8 7.4 36.8s-24.5 103.8-29 123.2c-5 21.4-3 51.6-.9 71.2C65.4 450.9 0 361.1 0 256 0 119 111 8 248 8s248 111 248 248z"
                    />
                  </svg>
                </button>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </article>
</div>
`;
